src/kernels/tensor_view.hpp

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GUARD_TENSOR_VIEW_H
+#define GUARD_TENSOR_VIEW_H
+
+template <int N>
+struct tensor_layerout_t;
+
+template <int N>
+struct tensor_view_t
```

Comment:
overall this design is more robust than defining macros (as previously).

Reply:
Hi, I want to discuss something about tensor_view implementation. This file is highly related to getitem/utils.hpp. We need to call get_inner_expanded_tv function in utils.hpp to get tensor_view from a TensorDescriptor. However, I saw that utils.hpp is created inside src/include/miopen/getitem directory. So that means in the future, when someone add new operations that requires tensor_view, for example operation XYZ, they also need to create a file utils.hpp with function get_inner_expanded_tv inside a new folder src/include/miopen/XYZ, which is the same as getitem/utils.hpp. I think this will create many duplicate files. So I think we should put get_inner_expanded_tv to tensor_view.hpp, delete utils.hpp, so that when creating new operation in the future that requires tensor_view, we can just simply call function get_inner_expanded_tv inside tensor_view.hpp instead of creating a new file utils.hpp